File: final/server/src/main/java/com/google/maps/gaming/zoinkies/ReferenceData.java
package com.google.maps.gaming.zoinkies;

import java.util.ArrayList;
import java.util.List;

/**
 * Types of reference data
 * - Location Objects
 * - Collectibles
 * - Weapons
 * - Armors
 * - Characters
 */
public class ReferenceData {

  private List<ReferenceItem> References;

  public ReferenceData() {
    References = new ArrayList<>();
  }

  public List<ReferenceItem> getReferences() {
    return References;
  }

  public void setReferences(List<ReferenceItem> references) {
    References = references;
  }

  public ReferenceItem getReferenceItem(String id) {
    if (id == null || id.isEmpty()) {
      return null;
    }
    for(ReferenceItem ri:getReferences()) {
      if (ri.getId().equals(id)) {
        return ri;
      }
    }
    return null;
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    for (ReferenceItem ri : References ) {
      sb.append(ri.toString());
      sb.append("\n");
    }
    return sb.toString();
  }
}

File: part3/client/Zoinkies/Assets/Zoinkies/Scripts/Models/ReferenceData.cs
/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
using System.Collections.Generic;
using System.Text;

namespace Google.Maps.Demos.Zoinkies {

/// <summary>
/// Keeps track of reference data for the game.
///
/// Types of reference data:
/// - Location Objects
/// - Collectibles
/// - Weapons
/// - Armors
/// - Characters
/// </summary>
  public class ReferenceData {

    /// <summary>
    /// List of all references used in the game.
    /// </summary>
    public List<ReferenceItem> references { get; set; }

    public ReferenceData() {
      references = new List<ReferenceItem>();
    }

    public override string ToString() {
      var sb = new StringBuilder();
      foreach (ReferenceItem ri in references) {
        sb.Append(ri);
        sb.Append("\n");
      }

      return sb.ToString();
    }
  }
}
